### PR TITLE
Added PCA

### PR DESCRIPTION
**Pull request by utkarsh0702**

### CODE REVIEW

**Comment by rohansingh9001** [comment]
@utkarsh0702 there are some build errors in your code which corresponds to wrong formatting which we require. Please use tools like flake8 to correct the formatting or refer to this error log [here](https://travis-ci.org/github/RoboticsClubIITJ/ML-DL-implementation/builds/748758301).

**Comment by kwanit1142** [comment]
@utkarsh0702 apart from format errors mentioned by @rohansingh9001 , Why you used scipy library? Also, provide example to show the working of PCA, alongside input dataset. For reference, see other model implementation examples in this Repository

**Comment by utkarsh0702** [comment]
@rohansingh9001 I will go through the errors.
@kwanit1142 Ok, I will do so. I have used scipy for the gammaln function.

**Comment by kwanit1142** [comment]
@utkarsh0702 avoid using libraries other than Matplotlib and Numpy . Instead of using Scipy, try to make utility file of the function required.

**Comment by utkarsh0702** [comment]
@kwanit1142 I will try doing so

**Comment by utkarsh0702** [comment]
Can we add 3 more loss functions: Mean Bias Error, Hinge Loss and Cross-Entropy Loss?
If possible can I work on them?

**Comment by kwanit1142** [comment]
@utkarsh0702 this implementation is now better than before, but there is one issue u would have to solve, i.e Instead of importing pandas, try to change file type, like in txt format or so, such that numpy can easily fetch it, Otherwise LGTM...

**Comment by utkarsh0702** [comment]
@kwanit1142 I will see to it.

**Comment by utkarsh0702** [comment]
@kwanit1142 Is it fine?

**Comment by kwanit1142** [comment]
@utkarsh0702 regarding point-system queries, @rohansingh9001 can clarify to you. In addition to this, he can review this pull request too. Otherwise, LGTM for now.

**Comment by utkarsh0702** [comment]
@rohansingh9001 Please review the pr.
Please tell me how many points are allotted to a person for resolving an issue in NWoC?

**Comment by utkarsh0702** [comment]
@kwanit1142 Can we add 3 more loss functions: Mean Bias Error, Hinge Loss and Cross-Entropy Loss?
If possible can I work on them?

**Review by kwanit1142 — APPROVED** [review]

**Comment by kwanit1142** [comment]
Looks good to Me Now..
